hack/table-gen/table-gen.go

```diff
@@ -0,0 +1,264 @@
+	description := ""
+	if d := element["description"]; d != nil {
+		description = d.(string)
+	}
+
+	fullName := fmt.Sprintf("%s%s", parentPath, name)
+	skipWithAncestors, shouldBeSkipped := elementsToSkip[fullName]
+	if shouldBeSkipped && skipWithAncestors {
+		return result
+	}
+
+	if !shouldBeSkipped {
+		result[fullName] = generateTableRow(fullName, description, name)
+	}
+
+	if elementType == "object" {
+		mergeMaps(result, generateObjectDoc(elementsToSkip, element, name, parentPath))
+	}
+
+	if elementType == "array" {
+		mergeMaps(result, generateArrayDoc(elementsToSkip, element, name, parentPath))
+	}
+
+	return result
+}
+
+// generates table row out of CRD object with type object. Returns resulting map with path of an element and table row
+func generateObjectDoc(elementsToSkip map[string]bool, element map[string]interface{}, name string, parentPath string) map[string]string {
+	result := map[string]string{}
+	properties := getElement(element, "properties")
+	if properties == nil {
+		return result
+	}
+
+	propMap := properties.(map[string]interface{})
+	for _, propName := range sortKeys(propMap) {
+		mergeMaps(result, generateElementDoc(elementsToSkip, propMap[propName], propName, parentPath+name+"."))
+	}
+	return result
+}
+
+// generates table row out of CRD object with type array. Returns resulting map with path of an element and table row
+func generateArrayDoc(elementsToSkip map[string]bool, element map[string]interface{}, name string, parentPath string) map[string]string {
+	result := map[string]string{}
+	properties := getElement(element, "items")
+	if properties == nil {
+		return result
+	}
+
+	description := ""
+
+	if element["description"] != nil {
+		description = element["description"].(string)
+	}
+
+	propMap := properties.(map[string]interface{})
+
+	if description == "" && propMap["description"] != nil {
+		description = propMap["description"].(string)
+	}
+
+	result = generateObjectDoc(elementsToSkip, propMap, name, parentPath)
+
+	result[parentPath+name] = generateTableRow(parentPath+name, description, name)
+
+	return result
+}
+
+// generates a row of the resulting table which we include into our MD file
+func generateTableRow(fullName string, description string, name string) string {
+	return fmt.Sprintf("| **%s** | %s |",
+		fullName, normalizeDescription(description, name))
+}
+
+// goes through the map, and returns a specific element based on a path we provided
+func getElement(obj interface{}, path ...string) interface{} {
+	elem := obj
+	for _, p := range path {
+		elem = elem.(map[string]interface{})[p]
+	}
+	return elem
+}
+
+// transforms description, so it would be a better fit for the table
+func normalizeDescription(description string, name string) string {
+	description_trimmed := strings.Trim(description, " ")
+	name_trimmed := strings.Trim(name, " ")
+	if len(name_trimmed) == 0 {
+		return description_trimmed
+	}
+	dParts := strings.SplitN(description_trimmed, " ", 2)
```

Comment:
I can't understand what we are doing here in this function starting from this line. Can you add comments to clarify what we are doing here? why do we need to split the description?